src/MongoDB.Driver.Core/Core/Compression/Snappy/Snappy32NativeMethods.cs

```diff
@@ -0,0 +1,108 @@
+        private static readonly Lazy<Delegates32.snappy_max_compressed_length> __snappy_max_compressed_length;
+        private static readonly Lazy<Delegates32.snappy_uncompress> __snappy_uncompress;
+        private static readonly Lazy<Delegates32.snappy_uncompressed_length> __snappy_uncompressed_length;
+        private static readonly Lazy<Delegates32.snappy_validate_compressed_buffer> __snappy_validate_compressed_buffer;
+
+        // static constructor
+        static Snappy32NativeMethods()
+        {
+            var snappyLocator = new SnappyLocator();
+            __libraryLoader = new Lazy<LibraryLoader>(() => new LibraryLoader(snappyLocator), isThreadSafe: true);
+
+            __snappy_compress = CreateLazyForDelegate<Delegates32.snappy_compress>(nameof(snappy_compress));
+            __snappy_max_compressed_length = CreateLazyForDelegate<Delegates32.snappy_max_compressed_length>(nameof(snappy_max_compressed_length));
+            __snappy_uncompress = CreateLazyForDelegate<Delegates32.snappy_uncompress>(nameof(snappy_uncompress));
+            __snappy_uncompressed_length = CreateLazyForDelegate<Delegates32.snappy_uncompressed_length>(nameof(snappy_uncompressed_length));
+            __snappy_validate_compressed_buffer = CreateLazyForDelegate<Delegates32.snappy_validate_compressed_buffer>(nameof(snappy_validate_compressed_buffer));
+        }
+
+        // public static methods
+        public static SnappyStatus snappy_compress(IntPtr input, uint input_length, IntPtr output, ref uint output_length)
+        {
+            return __snappy_compress.Value(input, input_length, output, ref output_length);
+        }
+
+        public static uint snappy_max_compressed_length(uint input_length)
+        {
+            return __snappy_max_compressed_length.Value(input_length);
+        }
+
+        public static SnappyStatus snappy_uncompress(IntPtr input, uint input_length, IntPtr output, ref uint output_length)
+        {
+            return __snappy_uncompress.Value(input, input_length, output, ref output_length);
+        }
+
+        public static SnappyStatus snappy_uncompressed_length(IntPtr input, uint input_length, out uint output_length)
+        {
+            return __snappy_uncompressed_length.Value(input, input_length, out output_length);
+        }
+
+        public static SnappyStatus snappy_validate_compressed_buffer(IntPtr input, uint input_length)
+        {
+            return __snappy_validate_compressed_buffer.Value(input, input_length);
+        }
+
+        // private static methods
+        private static Lazy<TDelegate> CreateLazyForDelegate<TDelegate>(string name)
+        {
+            return new Lazy<TDelegate>(() => __libraryLoader.Value.GetDelegate<TDelegate>(name), isThreadSafe: true);
+        }
+
+        // nested types
+        private class Delegates32
+        {
+            [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
```

Comment:
See for details: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/82#discussion_r347503487